### DISCUSSION TITLE
Should A11y Settings appear at the toplevel?

### DISCUSSION THREAD

**Discussion by danirabbit** [elementary/quick-settings#21] [Ideas]
It's not clear to me which a11y settings are things that people actually quick toggle and if those things should appear at the toplevel instead of in a subpage.

![Screenshot from 2024-02-12 16 37 38](https://github.com/elementary/quick-settings/assets/7277719/c8bab1d9-722f-4d5d-9993-094e4038cc5a)

For example should screen reader and onboard keyboard items appear next to toggles like dark mode and rotation lock? Does it make more sense for them to be at the top level instead of in a sub page?

![Screenshot from 2024-02-13 19 31 51](https://github.com/elementary/quick-settings/assets/7277719/0b413a04-811e-4f61-98dd-073af9570db4)

What about the text size slider? Is this even something people expect to be changing often enough for it to appear here at all?

Are the rest of these toggles like slow and bounce keys things that people regularly toggle on and off or should these also not be present in the indicator? I think the scope of the indicator should be settings that folks are toggling on and off multiple times a session so they need quick access. If this is a set-and-forget setting, we should not show it here

**Comment by CallMeAl1973** [comment]
As someone who exclusively uses a screen reader, the actual location of the toggle  is honestly not very relevant to me. I say that because the ability to enable the screen reader is so important that I can't even locate the toggle unless I already have access to the screen reader.
Does Elementary make use of the alt+super+s keystroke to toggle the screen reader? I think for the most important accessibility functions, such as screen reader, magnifier, sticky keys, maybe others, these toggles need to be available globally if at all possible. Examples: Alt+super+s for screen reader, maybe hold down a modifier for 5 seconds or more to enable sticky keys, some other alt+super key combination for magnifier ...
if all the important toggles were available as global keystrokes, then you could have much more flexibility in terms of the layout of the actual UI.

**Reply by danirabbit** [reply]
Yes we use the Super + Alt + S keyboard shortcut for the screen reader and that shortcut is accessible during the installer session and on the lock screen as well :)

This discussion post is more about asking which of these settings should *also* be available from Quick Settings and which of these things aren't settings that one would be toggling on and off throughout the session

**Comment by RasheedAZ** [comment]
Thanks for taking all this into account and asking. I think easy, fast access to the text size slider is very useful. I use the Large Text toggle in Gnome regularly.

Letting the user decide which ones appear there would be better but probably too much work and not worth the effort...?

Also, something is general: if anything could be enabled by default by an accidental key combo, it'd be really useful if the toggle indicating that was easily accessible or else better let the user decide which key combos they want to enable. This is one reason I absolutely dislike gestures and disable them because like unused shortcut key combos, they do things the user never intended.

**Reply by danirabbit** [reply]
Thanks for that feedback about the text size! Could you talk a little bit more about how you use this setting? I'd like to make sure that I'm providing the most helpful version of this as possible

Yeah I think long term it would be better to individually choose which toggles to include in QuickSettings. At the moment there is a single option to show all/none a11y-related options.

**Reply by RasheedAZ** [reply]
You're welcome. I usually need quick access to this settings in 2 cases:
- certain apps whose text is small with no easy way of changing it
- making text larger quickly for screenshare during online teaching

**Comment by Oowoosh0** [comment]
What settings you use so often that you would want them on the top level of a quick settings menu is probably very different for everyone. Personally I've never used the onscreen-keyboard or the screenreader but of course there are people who use those on a daily basis. And while I switch from dark to light mode multiple times a day someone else might choose one an stick to it.
So maybe a model where the user can choose themself which options they want on the top level of the menu would be the best.
That would be similar to what you currently have in Android, where the pull-down quick settings menu is populated with a default selection of settings that can be edited by the user if they want.

**Reply by teamcons** [reply]
I agree. Accessibility however is a different beast
I cannot recall if the first start menu/app offers some accessibility options ? (i recall i had to look around for bigger fonts and interface, but im also impatient)
It could be interesting to have the "enable once and done" accessibility settings there, and include in it a toggle for "include in quick menu these accessibility settings because damn i really need to tinker with that shit regularly"
However it would just displace the main question

**Reply by danirabbit** [reply]
There is currently already a setting for showing/hiding a11y settings in the panel. That still exists so you do have the option to hide/show this group of settings from the Quick Settings menu as well. I do agree that in the end we probably want more granular customization for quick settings as in iOS and Android.

So the scope of this discussion is more about: for people who do enable this setting currently, which of these options are actually helpful and which of them are just noise.

Also, @teamcons yes we make sure to show a11y options during the installer and greeter sessions!
